datalad/support/annexrepo.py

```diff
@@ -382,29 +382,66 @@ def annex_get(self, files, log_online=True, options=None):
                                 log_stdout=True, log_stderr=not log_online,
                                 log_online=log_online, expect_stderr=True)
 
-    # TODO: Moved from HandleRepo. Just a temporary alias.
-    # When renaming is done, melt with annex_get
-    def get(self, files):
-        """get the actual content of files
-
-        This command gets the actual content of the files in `list`.
-        """
-        self.annex_get(files)
-
     @normalize_paths
-    def annex_add(self, files, backend=None, options=None):
-        """Add file(s) to the annex.
+    def add(self, files, git=False, backend=None, options=None, commit=False,
+            msg=None):
+        """Add file(s) to the repository.
+
 
         Parameters
         ----------
         files: list of str
-            list of paths to add to the annex
+          list of paths to add to the annex
+        git
+        commit
+        msg
+        backend
+        options
+
+        Returns
+        -------
+        list of dict
         """
-        options = options[:] if options else []
+        # Note: As long as we support direct mode, one should not call
+        # super().add() directly. Once direct mode is gone, we might remove
+        # `git` parameter and call GitRepo's add() instead.
+        if git:
+            # add to git instead of annex
+            # TODO: `options` currently unused in case of git
+            if self.is_direct_mode():
+                cmd_list = ['git', '-c', 'core.bare=false', 'add'] + files
```

Comment:
pardon me but we do need a unittest for this core functionality... should really trivial to do... could even be the same test just ran in both direct and indirect modes -- we have already a few like that